From a46d45305c1dbf57e8da9ba2108ca3e38bccda2d Mon Sep 17 00:00:00 2001
From: Emilian Bold <emi@apache.org>
Date: Tue, 6 Nov 2018 21:38:29 +0200
Subject: [PATCH 11/32] Do not open Exception Reporter if the meta refresh url
 is empty
License: Copyright (c) 2018, 2019 Emilian Marius Bold
 
 Permission to use, copy, modify, and distribute this software for any
 purpose with or without fee is hereby granted, provided that the above
 copyright notice and this permission notice appear in all copies.
 
 THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
 WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
 MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
 ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
 WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
 ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.

---
 .../src/org/netbeans/modules/uihandler/Installer.java         | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/platform/uihandler/src/org/netbeans/modules/uihandler/Installer.java b/platform/uihandler/src/org/netbeans/modules/uihandler/Installer.java
index 8070e29f27..56bc60938b 100644
--- a/platform/uihandler/src/org/netbeans/modules/uihandler/Installer.java
+++ b/platform/uihandler/src/org/netbeans/modules/uihandler/Installer.java
@@ -1416,6 +1416,10 @@ public class Installer extends ModuleInstall implements Runnable {
             assert m != null;
             if (m.find()) {
                 LOG.log(Level.FINE, "uploadLogs, found url = {0}", m.group(1)); // NOI18N
+                if (m.group(1).isEmpty()) {
+                    //EMI: Allow some way of not redirecting!
+                    return null;
+                }
                 return new URL(m.group(1));
             } else {
                 assert redir != null;
-- 
2.21.0

